app/src/split/bluetooth/central.c

```diff
@@ -265,6 +270,95 @@ static uint8_t split_central_notify_func(struct bt_conn *conn,
     return BT_GATT_ITER_CONTINUE;
 }
 
+#if IS_ENABLED(CONFIG_ZMK_SPLIT_BLE_CENTRAL_BATTERY_LEVEL_FETCHING)
+
+static uint8_t peripheral_battery_levels[ZMK_SPLIT_BLE_PERIPHERAL_COUNT] = {0};
+
+int zmk_split_get_peripheral_battery_level(uint8_t source, uint8_t *level) {
+    if (source < 0 || source >= ZMK_SPLIT_BLE_PERIPHERAL_COUNT) {
```

Comment:
`source` is unsigned, so it can't be less than 0.

Also, consider using `ARRAY_SIZE(peripheral_battery_levels)` so the bounds check stays in sync with the size of the array it's checking, even if that size is changed to use a different symbol later.

Reply:
Good call. changed.